plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/MockAmazonS3.java

```diff
@@ -48,12 +50,30 @@
 
 class MockAmazonS3 extends AbstractAmazonS3 {
 
+    private final int mockSocketPort;
+
     private Map<String, InputStream> blobs = new ConcurrentHashMap<>();
 
     // in ESBlobStoreContainerTestCase.java, the maximum
     // length of the input data is 100 bytes
     private byte[] byteCounter = new byte[100];
 
+
+    MockAmazonS3(int mockSocketPort) {
+        this.mockSocketPort = mockSocketPort;
+    }
+
+    // simulate a socket connection to check that SocketAccess is used correctly
```

Comment:
Can you elaborate a bit more why this necessary, exactly what we are simulating here?